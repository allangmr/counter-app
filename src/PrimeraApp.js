import PropTypes from 'prop-types'
const PrimeraApp = ( {saludos,subtitulo} ) => {

    // Mala forma de validar que venga la variable
    // if(!saludos){
    //     throw new Error("The saludos is neccesary.")
    // }

    return (
        <>
            <h1>{saludos}</h1>
            {/* <pre>{JSON.stringify(saludo, null,3)}</pre> */}
            <p>{subtitulo}</p>
        </>
    );

}

PrimeraApp.propTypes = {
    saludos: PropTypes.string.isRequired
}
PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}
export default PrimeraApp;
