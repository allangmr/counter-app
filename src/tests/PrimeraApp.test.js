import PrimeraApp from "../PrimeraApp";
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'
describe('Test in PrimerApp.js', () => {
    // test('should be show the message Hello...', () => {
    //     //Arrange
    //     const saludo = 'Hola Soy Allan';
    //     //Act
    //     const { getByText } = render(<PrimeraApp saludos={saludo}/>)
    //     //Assert
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })
    //Compara 2 string el que se creo al momento del primer snapshot y el actual.
    test('should be show <PrimeraApp /> ', () => {
        const saludo = 'Hola Soy Allan';
        const wrapper = shallow(<PrimeraApp saludos={saludo} />);

        expect(wrapper).toMatchSnapshot();
    })
    //compara el props y el resultado del envio del props al componente 
    test('should be show the subtitle send by props', () => {
        const saludo = 'Hola Soy Allan';
        const subtitulo = 'No tenemos subtitulos';
        const wrapper = shallow(<PrimeraApp 
            saludos={saludo} 
            subtitulo={subtitulo}
            />);

        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(subtitulo);
    })
    
    
    
});