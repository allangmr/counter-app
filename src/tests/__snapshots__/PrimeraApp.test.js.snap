// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in PrimerApp.js should be show <PrimeraApp />  1`] = `
<Fragment>
  <h1>
    Hola Soy Allan
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
